### PR TITLE
DEPR: Change str.replace(regex) from True to False & single behavior

### LINKED ISSUE #24804: str.replace('.','') should replace every character?
#### Code Sample (pandas 0.23.0)

In [1]:
```
import pandas as pd
s = pd.Series(['abc','123'])
s.str.replace('.','',regex = True)
```
Out [1]:
```
0    abc
1    123
dtype: object
```

#### Problem description

Hi everyone, I was showing str.replace to a colleague and how it uses regex by default and when I entered pd.Series.str.replace('.','') I expected every character to be removed but instead nothing happened. Is that a bug or is there something I don't understand?

#### Expected Output: empty strings ""
Out [1]:
```
0    
1    
dtype: object
```

#### Output of pandas 0.23.0

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 85 Stepping 4, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.0
pytest: 3.5.1
pip: 18.0
setuptools: 39.1.0
Cython: 0.28.2
numpy: 1.15.4
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 7.1.1
sphinx: 1.7.4
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: 1.2.1
tables: 3.4.3
numexpr: 2.6.5
feather: None
matplotlib: 2.2.2
openpyxl: 2.4.11
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.4
lxml: 4.2.3
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.1.18
pymysql: None
psycopg2: 2.7.5 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #24804 (Replace xxxx with the GitHub issue number)

Introduced in #36695

### CODE REVIEW

File: doc/source/user_guide/text.rst [hunk 1]
```diff
@@ -267,14 +267,13 @@ i.e., from the end of the string to the beginning of the string:
    s3
    s3.str.replace("^.a|dog", "XX-XX ", case=False, regex=True)
 
-.. warning::
+Single character pattern with ``regex=True`` will also be treated as regular expressions:
```
Comment (jbrockmendel) [inline doc/source/user_guide/text.rst:270]: this line seems weird without the context that it didn't use to be the case

Comment (mroeschke) [reply doc/source/user_guide/text.rst:270]: Fair point; will add a `.. versionchanged 2.0` here

**Review by jbrockmendel — APPROVED** [review]
LGTM pending greenish

**Comment by jbrockmendel** [comment]
updated #30228?

**Comment by mroeschke** [comment]
> updated #30228?

Thanks for the reminder. Updated

**Pull request merged**